src/NuGet.Core/NuGet.Commands/RestoreCommand.cs

```diff
@@ -68,7 +68,7 @@ public Task<RestoreResult> ExecuteAsync()
         public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
         {
             // Use the shared cache if one was provided, otherwise create a new one.
-            var localRepository = _request.SharedLocalCache ?? new NuGetv3LocalRepository(_request.PackagesDirectory);
+            var localRepository = _request.SharedCache.LocalCache ?? new NuGetv3LocalRepository(_request.PackagesDirectory);
```

Comment:
1. should we enforce the requirement to pass a cache in? What is the scenario where we create a new one here?
2. If you disagree with 1, Can the sharedCache be null